app/callbacks/display_callbacks.py

```diff
@@ -268,11 +225,8 @@ def update_image_and_bbox(slider_value, alert_data, media_url, alert_list):
     images = []
     boxes = []
     for _, alert in alert_data.iterrows():
-        event_id = str(alert["event_id"])
-        media_id = str(alert["media_id"])
-        if event_id in media_url and media_url[event_id].get(media_id, "").strip():
-            images.append(media_url[event_id][media_id])
-            boxes.append(alert["processed_loc"])
+        images.append(alert["media_url"])
+        boxes.append(alert["processed_loc"])
```

Comment:
I know it doesn't sound obvious, but this will be faster & require less RAM:
```python
images, boxes = zip(*[(alert["media_url"], alert["processed_loc']) for _, alert in alert_data.iterrows()]) 
```

Reply:
thnaks !